_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -85,6 +87,141 @@ bool CheckHardwareSupport(VideoCORE *p_core, mfxVideoParam *p_video_param)
     return true;
 }
 
+class ReferenceFrameStorage
+{
+private:
+    class FrameLocker
+    {
+    private:
+        UMC::FrameMemID m_frameId = -1;
+        std::shared_ptr<mfx_UMC_FrameAllocator> m_frameAllocator;
```

Comment:
raw pointer is enough, please avoid ownership sharing if it's not really required by design

Reply:
Done.